### PR TITLE
Test scenario for adding event group to an event

### PR DESCRIPTION
**Pull request by mp-sgent**
image

### CODE REVIEW

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 1]
```diff
@@ -237,6 +361,13 @@ public EventDirectorySteps(
           webDriverHelpers.clickOnWebElementBySelector(APPLY_FILTERS_BUTTON);
           TimeUnit.SECONDS.sleep(10);
         });
+    When(
+        "I hover to Event Groups column of the Event result",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:365]: what is the need for this hover?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 2]
```diff
@@ -57,6 +99,81 @@ public EventDirectorySteps(
               dataOperations.getPartialUuidFromAssociatedLink(eventUuid));
         });
 
+    When(
+        "I click checkbox to choose all Event results",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:103]: add location for this action

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 3]
```diff
@@ -57,6 +99,81 @@ public EventDirectorySteps(
...
+    When(
+        "I navigate to the last created Event page via URL",
+        () -> {
+          String createdEventUUID = CreateNewEventSteps.newEvent.getUuid();
+          String LAST_CREATED_EVENT_PAGE_URL =
+              environmentUrl + "/sormas-webdriver/#!events/data/" + createdEventUUID;
+          webDriverHelpers.accessWebSite(LAST_CREATED_EVENT_PAGE_URL);
+          webDriverHelpers.waitForPageLoaded();
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(UUID_INPUT, 50);
+          TimeUnit.SECONDS.sleep(5);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:124]: if the above method wait is correct, why do we need the sleep?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 4]
```diff
@@ -57,6 +99,81 @@ public EventDirectorySteps(
...
+        });
+
+    When(
+        "^I click on ([^\"]*) Radiobutton on Event Directory Page$",
+        (String buttonName) -> {
+          webDriverHelpers.clickWebElementByText(EVENTS_RADIO_BUTTON, buttonName);
+          webDriverHelpers.waitForPageLoaded();
+        });
+    When(
+        "I check that data of linked group is correct",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:134]: let's be a bit more specific in the name to know exactly what we're comparing

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 5]
```diff
@@ -57,6 +99,81 @@ public EventDirectorySteps(
...
+        () -> {
+          EventGroup createdGroup = EditEventSteps.groupEvent;
+          EventGroup collectedGroup = collectEventGroup();
+          ComparisonHelper.compareEqualFieldsOfEntities(
+              collectedGroup, createdGroup, List.of("name"));
+          webDriverHelpers.waitForPageLoaded();
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:140]: why to we need this if we compare 2 collected entities?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 6]
```diff
@@ -96,6 +213,13 @@ public EventDirectorySteps(
           webDriverHelpers.waitForPageLoaded();
           webDriverHelpers.selectFromCombobox(FILTER_BY_DISEASE, disease);
         });
+    When(
+        "I filter by last created group in Event Directory Page",
+        () -> {
+          webDriverHelpers.waitForPageLoaded();
+          webDriverHelpers.fillInWebElement(
+              By.id("freeTextEventGroups"), EditEventSteps.groupEvent.getUuid());
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:221]: created proper locator

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java [hunk 7]
```diff
@@ -287,4 +427,10 @@ public EventDirectorySteps(
                         number.intValue(),
                         "Number of displayed cases is not correct")));
   }
+
+  private EventGroup collectEventGroup() {
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:431]: can we collect all data? in this way we can reuse this method for other tests, not only for one field needed here.
If we don't need to collect a POJO, then we can scrap data from this field and verify it directly in step

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java (hunk 5 above)
Comment (mp-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:140]: removed collectEventGroup() and included getting name of linked group to collectedGroup object. We are comparing only names for this test step.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java (hunk 1 above)
Comment (mp-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:365]: It is required in test case to hover to that localization.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java (hunk 7 above)
Comment (mp-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java:431]: removed it and moved collection of the Event group name to required location.

**Comment by mp-sgent** [comment]
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/46/parameters/

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**